### PR TITLE
C++ API parity: at::Tensor::version

### PR DESCRIPTION
**Pull request by pbelevich**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #26332 C++ API parity: at::Tensor::requires_grad_
* #26251 C++ API parity: at::Tensor::detach
* **#26217 C++ API parity: at::Tensor::version**

Differential Revision: [D17427577](https://our.internmc.facebook.com/intern/diff/D17427577)

### CODE REVIEW

**Comment by yf225** [comment]
@pbelevich Sorry I just realized we can actually name the method `_version` instead of `version` to better match the Python API, which should also fix the CI test error. The reason I didn't suggest the `_version` name originally was because having a C++ identifier beginning with an underscore is usually not encouraged, but since we already have a lot of functions that start with an underscore (e.g. `func: _copy_from`) in native_functions.yaml, it might be acceptable to do so here.

**Comment by pbelevich** [comment]
@yf225 [def check_methods_do_not_start_with_underscore](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/function_wrapper.py#L705-L713) prevents me from doing this

**Comment by yf225** [comment]
@pbelevich for `check_methods_do_not_start_with_underscore`, we can make an exception for this case and add `_version` to the whitelist

**Review by yf225 — APPROVED** [review]
Looks awesome, thanks a lot @pbelevich ! I re-ran the failed CI jobs so that we can get the full set of signals.
